web-local/src/routes/(misc)/deploy/+page.svelte

```diff
@@ -31,6 +32,23 @@
   onMount(() => {
     void deployer.loginOrDeploy();
   });
+
+  let upgradeHref = "";
+  $: if ($org && $metadata.data) {
+    let cloudUrl = $metadata.data.adminUrl.replace(
+      "admin.rilldata",
+      "ui.rilldata",
+    );
+    // hack for dev env
+    // TODO: find a better solution and get a url from backend
+    if (cloudUrl === "http://localhost:9090") {
+      cloudUrl = "http://localhost:3000";
+    }
+    const url = new URL(cloudUrl);
+    url.pathname = `/${$org}/-/settings/billing`;
+    url.searchParams.set("upgrade", "true");
+    upgradeHref = url.toString();
```

Comment:
Given there's some complexity here, it'd be nice to quarantine this in a utility function `makeUpgradeHref` with comments that make it more readable.